src/components/hooks/useTheme.js

```diff
@@ -0,0 +1,15 @@
+import React, { useMemo } from "react";
+import { useColorScheme } from "react-native";
+
+const useTheme = (__styles) => {
+  const colorScheme = useColorScheme();
+  const isDark = colorScheme === "dark";
+
+  const styles = useMemo(() => {
+    return __styles(isDark);
+  }, [colorScheme]);
```

Comment:
Here the list of dependencies should be:
```suggestion
  }, [__styles, isDark]);
```
because the memoized value need to be recomputed if any of `__styles` or `isDark` changes.

If you inline the `colorScheme === "dark"`, then the dependency can correctly include `colorScheme`.